exp/query/api.go

```diff
@@ -66,9 +68,11 @@ type API struct {
 
 	schema  SchemaQuerier
 	offload *modsloads3.Module
-	// NOTE: This lock may be in-efficient if downloaded needed from multiple tenants
-	// TODO(kavi): Optimize based on some data later.
-	omu sync.Mutex
+	// cachedTenantLsmkvStores: tenantName -> tenantLsmkvStore (lsm object and path to tenant dir on disk)
+	// the tenantLsmkvStore value is updated if new tenant data versions are downloaded.
+	// Note that tenantLsmkvStore.localTenantPath points to the root of the tenant dir,
+	// not the lsm subdir.
+	cachedTenantLsmkvStores sync.Map
```

Comment:
I would also love some metrics about this count. Just to see the memory footprint of this unbounded cardinality at scale. But something for follow up PR :)